File: src/sdk/android/3/core-classes/security/search-users/snippets/search-users-1.java

JSONObject filter = new JSONObject()
  .put("bool", new JSONObject()
    .put("must", new JSONArray()
      .put(new JSONObject()
        .put("terms", new JSONObject()
          .put("profileIds", new JSONArray().put("anonymous").put("default"))
        )
      )
      .put(new JSONObject()
        .put("geo_distance", new JSONObject()
          .put("distance", "10km")
          .put("pos", new JSONObject()
            .put("lat", 48.8566140)
            .put("lon", 2.352222)
          )
        )
      )
    )
  );

// optional: result pagination configuration
Options options = new Options();
options.setFrom((long) 0);
options.setSize((long) 42);
options.setScroll("1m");

kuzzle
  .security
  .searchUsers(filters, options, new ResponseListener<SecurityDocumentList>() {
    @Override
    public void onSuccess(SecurityDocumentList users) {
      // users.getDocuments() returns an users list
      for(User user : users.getDocuments()) {

      }

      // Total number of profiles, regardless of pagination
      long total = users.getTotal();

      // Available only if a "scroll" option has been provided
      String scrollId = users.getScroll()
    }

    @Override
    public void onError(JSONObject error) {

    }
  });

File: src/sdk/php/3/core-classes/security/search-users/snippets/search-users-1.php

<?php

use \Kuzzle\Kuzzle;
use \Kuzzle\Security\User;
use \Kuzzle\Util\UsersSearchResult;

$filters = [
  'bool' => [
    'must' => [
      [
        'terms' => [
          'profileIds' => ['anonymous', 'default'],
        ]
      ],
      [
        'geo_distance' => [
          'distance' => '10km',
          'pos' => [
            'lat' => 48.8566140,
            'lon' => 2.352222
          ]
        ]
      ]
    ]
  ]
];

// optional: result pagination configuration
$options = [
  'from' => 0,
  'size' => 1,
  'scroll' => '1m'
];

$kuzzle = new Kuzzle('localhost');
$security = $kuzzle->security();

try {
  $result = $security->searchUsers($filters, $options);

  // $result instanceof UsersSearchResult

  foreach($result->getUsers() as $user) {
    // $user instanceof User
  }
}
catch (ErrorException $e) {

}
